app/controllers/comments_controller.rb

```diff
@@ -342,36 +342,34 @@ def create
-          else
-            flash[:comment_notice] = ts("Comment created!")
-          end
-          respond_to do |format|
-            format.html do
-              if request.referer&.match(/inbox/)
-                redirect_to user_inbox_path(current_user, filters: filter_params[:filters], page: params[:page])
-              elsif request.referer&.match(/new/)
-                # came here from the new comment page, probably via download link
-                # so go back to the comments page instead of reloading full work
-                redirect_to comment_path(@comment)
-              elsif request.referer == "#{root_url}"
-                # replying on the homepage
-                redirect_to root_path
-              elsif @comment.unreviewed? && current_user
-                redirect_to comment_path(@comment)
-              elsif @comment.unreviewed?
-                redirect_to_all_comments(@commentable)
-              else
-                redirect_to_comment(@comment, {view_full_work: (params[:view_full_work] == "true"), page: params[:page]})
-              end
+        flash[:comment_notice] = if @comment.unreviewed?
+                                   # i18n-tasks-use t("comments.create.success.moderated.admin_post")
+                                   # i18n-tasks-use t("comments.create.success.moderated.work")
+                                   t("comments.create.success.moderated.#{@comment.ultimate_parent.model_name.i18n_key}")
+                                 else
+                                   t("comments.create.success.not_moderated")
+                                 end
+        respond_to do |format|
+          format.html do
+            if request.referer&.match(/inbox/)
+              redirect_to user_inbox_path(current_user, filters: filter_params[:filters], page: params[:page])
+            elsif request.referer&.match(/new/) || (@comment.unreviewed? && current_user)
+              # If the referer is the new comment page, go to the comment's page
+              # instead of reloading the full work.
+              # If the comment is unreviewed and commenter is logged in, take
+              # them to the comment's page so they can access the edit and
+              # delete options for the comment, since unreviewed comments don't
+              # appear on the commentable.
+              redirect_to comment_path(@comment)
+            elsif request.referer == root_url
+              # replying on the homepage
+              redirect_to root_path
+            elsif @comment.unreviewed?
+              redirect_to_all_comments(@commentable)
+            else
+              redirect_to_comment(@comment, { view_full_work: (params[:view_full_work] == "true"), page: params[:page] })
             end
           end
-        else
-          # this shouldn't come up any more
-          flash[:comment_notice] = ts("Sorry, but this comment looks like spam to us.")
-          redirect_back_or_default(root_path)
```

Comment:
I took this out (as well as the initial `@comment.approved?` check) because Reviewdog was really unhappy with the nesting here, and it had me making so many changes to this action that, hey, what's another one. We're just going to assume that if a comment is saved, it passed the spam check.